### DISCUSSION TITLE
Jobs are randomly skiped or cancelled with unknown errors

### DISCUSSION THREAD

**Discussion by njzjz** [community/community#126469] [Actions] [answered]
### Select Topic Area

Bug

### Body

Starting yesterday, I have seen a lot of jobs randomly skipped or cancelled with unknown errors.

![image](https://github.com/community/community/assets/9496702/9e721807-935b-44fd-8290-e2ed8f2b42d3)

![image](https://github.com/community/community/assets/9496702/243e61b6-03c5-4ec6-8010-5a56bca1029f)

Example: https://github.com/deepmodeling/deepmd-kit/actions/runs/9293232817

**Comment by indiarasabarreto** [comment]
Because if it's not a failure and not cancelled it's either success() or skipped.

1. `if: github.event_name == 'push' && !failure() && !cancelled() `
2.  Same thing. Add the if: condition to make sure it would behave the same as the previous step, even though there might exist some sort of inheritance I didn't dig through.

**Reply by saltydk** [reply]
This is an issue with GHA, not his conditionals, it cancelled the build without anything happening inside of the run.

**Comment by VietND96** [comment]
+1, the same for me in #126539

**Comment by grandmas-favorite** [comment, marked as answer]
Hi everyone (@njzjz, @VietND96, @saltydk, and @indiarasabarreto)!

We had found an issue where small set of jobs are being canceled while they are in progress, but it should have been resolved a couple of hours ago.

Cheers!

**Comment by hanabi1224** [comment]
+1 I just experienced the same. Jobs with conditions `if: github.event.pull_request.draft == false` get skipped when the PR is no longer a draft.  Retry does not fix the issue and I have to convert the PR to draft and then back to make it work.

https://github.com/ChainSafe/forest/actions/runs/9367294602/job/25786761395?pr=4387

P.S. When all CI jobs pass in https://github.com/ChainSafe/forest/pull/4387  I see 20 CI job failures on my iOS Github app, refreshing does not help

**Reply by saltydk** [reply]
I doubt that is related to this issue.
